### PR TITLE
Refactored using fractions.Fraction class

### PR DESCRIPTION
**Pull request by jerryc05**
^

### CODE REVIEW

File: PyAngle/Angle.py [hunk 1]
```diff
@@ -1,373 +1,279 @@
...
-            self.__createByRadian(rad=rad)
-        # (x, y)
-        elif x is not None and y is not None:
-            self.__createByXY(x=x, y=y)
-        # invalid params
-        else:
-            # TODO: raise Exception
-            raise GeneratorExit
+    def __init__(self, deg: input_t):
+        self._deg = Fraction(deg)
```
Comment (gzn00417) [inline PyAngle/Angle.py:24]: delegates creation to factory methods

File: PyAngle/Angle.py [hunk 2]
```diff
@@ -1,373 +1,279 @@
...
         self.adjust()
 
-    def __createByDMS(self, d, m=0.0, s=0.0):
-        """Private Initiator For Degree, Minute and Second
+    @staticmethod
+    def from_dms(deg: input_t, minute: input_t = 0, sec: input_t = 0) -> 'Angle':
```
Comment (gzn00417) [inline PyAngle/Angle.py:28]: constrain types within int/float or input checking?

File: PyAngle/Angle.py [hunk 3]
```diff
@@ -1,373 +1,279 @@
...
-        self.minute = self.degree = 0
-        # Satisfy 0 <= s < 60
-        self.minute += self.second // 60
-        self.second %= 60
-        # Satisfy 0 <= m < 60
-        self.degree += self.minute // 60
-        self.minute %= 60
-        # Satisfy 0 <= d < 360
-        self.degree %= 360
+        while self._deg < 0:
```
Comment (gzn00417) [inline PyAngle/Angle.py:54]: what about %, doesn't it support mod?

File: PyAngle/Angle.py [hunk 4]
```diff
@@ -1,373 +1,279 @@
...
+        return math.radians(self.to_deg())
 
-    def toString(self, form="aaa°bbb′ccc″") -> (str):
-        """Convert angle into string of form(default: `"aaa°bbb′ccc″"`)\n
-        `aaa` is degree, `bbb` is minute, `ccc` is second, `DDD` is only degree(float),
-        `RRR` is radian, `XXX` is horizontal ordinate, `YYY` is vertical ordinate.\n
-        > default: `{:.2f}`\n
-        Eg.\n
-        >>> angle.toString()
+    def to_atan2(self, x=None, y=None) -> (float, float):
```
Comment (gzn00417) [inline PyAngle/Angle.py:160]: I was waiting for DALAO refactoring this ~~rubbish~~

File: PyAngle/AngleTest.py [hunk 5]
```diff
@@ -89,35 +89,34 @@ def test_cmp(self):
...
+        self.assertTrue(Angle.from_dms(deg=180).isStraightAngle())
+        self.assertFalse(Angle.from_dms(deg=190).isMinorAngle())
+        self.assertTrue(Angle.from_dms(deg=110).isMinorAngle())
+        self.assertTrue(Angle.from_dms(deg=200).isMajorAngle())
 
 
 if __name__ == "__main__":
-    unittest.main()
-
+    unittest.main()
```
Comment (gzn00417) [inline PyAngle/AngleTest.py:122]: Don't DALAO checked my silly test cases? (dog head

**Review by gzn00417 — COMMENTED** [review]
to DALAO

File: PyAngle/Angle.py (hunk 1 above)
Comment (jerryc05) [reply PyAngle/Angle.py:24]: just a must-exist fking stupid useless constructor

File: PyAngle/Angle.py (hunk 2 above)
Comment (jerryc05) [reply PyAngle/Angle.py:28]: does NOT constrain types. this is just a type HINT, and the code currently does not check for unsupported types

File: PyAngle/Angle.py (hunk 3 above)
Comment (jerryc05) [reply PyAngle/Angle.py:54]: ?

File: PyAngle/Angle.py (hunk 4 above)
Comment (jerryc05) [reply PyAngle/Angle.py:160]: submit an issue, or I will forget it. BTW who the F?ck is DALAO?

File: PyAngle/AngleTest.py (hunk 5 above)
Comment (jerryc05) [reply PyAngle/AngleTest.py:122]: You can handle test cases well. I trust you. I'm not DALAO, I'm a noob.

**Pull request merged**